dces-drc-integration/src/main/java/uk/gov/justice/laa/crime/dces/integration/aspect/CountedAspect.java

```diff
@@ -0,0 +1,40 @@
+package uk.gov.justice.laa.crime.dces.integration.aspect;
+
+import io.micrometer.core.instrument.Counter;
+import io.micrometer.core.instrument.MeterRegistry;
+import lombok.extern.slf4j.Slf4j;
+import org.aspectj.lang.JoinPoint;
+import org.aspectj.lang.annotation.After;
+import org.aspectj.lang.annotation.Aspect;
+import org.aspectj.lang.annotation.Pointcut;
+import org.springframework.stereotype.Component;
+
+@Aspect
+@Component
+@Slf4j
+public class CountedAspect {
+
+    private final MeterRegistry registry;
+
+    public CountedAspect(MeterRegistry registry) {
+        this.registry = registry;
+    }
+
+    @Pointcut("@within(Counted)")
+    public void countedClasses() {
+    }
+
+    @After("countedClasses()")
+    public void afterCountedMethod(JoinPoint joinPoint) {
+        String methodName = joinPoint.getSignature().toShortString();
+        log.info("Method name is {}", methodName);
+        Counter counter = Counter.builder("allAPIsCalls")
```

Comment:
I'm new to this and so this could be a stupid question, but is this building a new counter here every time? I think this will make a lot of counters each incremented just once, or am I missing something?